speechbrain/lobes/models/spacy/nlp.py

```diff
@@ -0,0 +1,95 @@
+"""Models and tooling for natural language processing using spaCy
+
+Authors
+* Sylvain de Langen 2024
+"""
+
+import spacy
+import spacy.tokens
+
+from typing import List, Union, Iterable
+
+
+def _as_sentence(sentence: Union[str, List[str]]):
+    """Ensures that a sentence is a `str` rather than a list of `str` tokens to
+    be passed to spaCy pipelines correctly.
+
+    Arguments
+    ---------
+    sentence: str or list of str
+        Sentence to return or list of tokens.
+
+    Returns
+    -------
+    str
+        The sentence, returned from the `sentence` argument as-is or joined with
+        spaces from a list of tokens."""
+
+    if isinstance(sentence, str):
+        return sentence
+
+    return " ".join(sentence)
+
+
+def _extract_lemmas(docs: Iterable[spacy.tokens.Doc]):
+    """Returns a batch of list of lemmas from a list of Doc (as returned by the
+    pipeline).
+
+    Arguments
+    ---------
+    docs: iterable of Doc
+        Documents, typically as returned by `nlp.pipe`.
+
+    Returns
+    -------
+    list of list of str
+        For each sentence, the sequence of extracted lemmas as `str`s."""
+    return [[tok.lemma_ for tok in doc] for doc in docs]
+
+
+class SpacyPipeline:
+    """Modelizes a spaCy pipeline."""
+
+    def __init__(self, nlp):
+        self.nlp = nlp
```

Comment:
missing docstring for input args. 

Wondering if we should add an example as well of how to use it.